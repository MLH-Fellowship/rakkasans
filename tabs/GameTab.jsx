import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import CountDown from 'react-native-countdown-component';
import { LinearGradient } from 'expo-linear-gradient';

import Colors from '../constants/Colors';

export default function GameTab() {
  const styles = StyleSheet.create({
    text: {
      fontSize: 40,
      color: Colors.white,
      marginBottom: 30,
    },
  });
  return (
    <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
      <LinearGradient
        colors={[Colors.white, Colors.primary]}
        style={{
          position: 'absolute',
          left: 0,
          right: 0,
          top: 0,
          height: 950,
        }}
      />
      <Text style={styles.text}>COMING SOON!</Text>
      <CountDown
        until={3000000}
        onFinish={() => alert('finished')}
        onPress={() => alert('coming soon...')}
        digitStyle={{ backgroundColor: '#FFF' }}
        digitTxtStyle={{ color: Colors.primary }}
        // separatorStyle={{ color: Colors.primary }}
        timeToShow={['D', 'H', 'M', 'S']}
        timeLabels={{
          d: 'Days', h: 'Hours', m: 'Mins', s: 'Secs',
        }}
        timeLabelStyle={{ color: Colors.white }}
        size={30}
      />
    </View>
  );
}
