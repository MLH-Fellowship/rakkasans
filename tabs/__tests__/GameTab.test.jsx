import React from 'react';
import renderer from 'react-test-renderer';

import GameTab from '../GameTab';

jest.useFakeTimers();

it('renders correctly', () => {
  const tree = renderer.create(<GameTab />).toJSON();
  expect(tree).toMatchSnapshot();
});
