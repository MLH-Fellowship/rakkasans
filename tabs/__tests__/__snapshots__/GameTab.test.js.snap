// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      Object {
        "colors": Array [
          4294967295,
          4278323777,
        ],
        "endPoint": undefined,
        "locations": undefined,
        "startPoint": undefined,
      }
    }
    style={
      Object {
        "height": 950,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#ffffff",
        "fontFamily": "fira-sans",
        "fontSize": 40,
        "marginBottom": 30,
      }
    }
  >
    COMING SOON!
  </Text>
  <View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 5,
                  "justifyContent": "center",
                  "marginHorizontal": 2,
                },
                Object {
                  "height": 78,
                  "width": 69,
                },
                Object {
                  "backgroundColor": "#FFF",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontVariant": Array [
                      "tabular-nums",
                    ],
                    "fontWeight": "bold",
                  },
                  Object {
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#020a41",
                    "fontFamily": "fira-sans",
                  },
                ]
              }
            >
              34
            </Text>
          </View>
        </View>
        <Text
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "white",
                "marginVertical": 2,
              },
              Object {
                "fontSize": 16.666666666666668,
              },
              Object {
                "color": "#ffffff",
                "fontFamily": "fira-sans",
              },
            ]
          }
        >
          Days
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 5,
                  "justifyContent": "center",
                  "marginHorizontal": 2,
                },
                Object {
                  "height": 78,
                  "width": 69,
                },
                Object {
                  "backgroundColor": "#FFF",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontVariant": Array [
                      "tabular-nums",
                    ],
                    "fontWeight": "bold",
                  },
                  Object {
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#020a41",
                    "fontFamily": "fira-sans",
                  },
                ]
              }
            >
              17
            </Text>
          </View>
        </View>
        <Text
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "white",
                "marginVertical": 2,
              },
              Object {
                "fontSize": 16.666666666666668,
              },
              Object {
                "color": "#ffffff",
                "fontFamily": "fira-sans",
              },
            ]
          }
        >
          Hours
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 5,
                  "justifyContent": "center",
                  "marginHorizontal": 2,
                },
                Object {
                  "height": 78,
                  "width": 69,
                },
                Object {
                  "backgroundColor": "#FFF",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontVariant": Array [
                      "tabular-nums",
                    ],
                    "fontWeight": "bold",
                  },
                  Object {
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#020a41",
                    "fontFamily": "fira-sans",
                  },
                ]
              }
            >
              20
            </Text>
          </View>
        </View>
        <Text
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "white",
                "marginVertical": 2,
              },
              Object {
                "fontSize": 16.666666666666668,
              },
              Object {
                "color": "#ffffff",
                "fontFamily": "fira-sans",
              },
            ]
          }
        >
          Mins
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 5,
                  "justifyContent": "center",
                  "marginHorizontal": 2,
                },
                Object {
                  "height": 78,
                  "width": 69,
                },
                Object {
                  "backgroundColor": "#FFF",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontVariant": Array [
                      "tabular-nums",
                    ],
                    "fontWeight": "bold",
                  },
                  Object {
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#020a41",
                    "fontFamily": "fira-sans",
                  },
                ]
              }
            >
              00
            </Text>
          </View>
        </View>
        <Text
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "white",
                "marginVertical": 2,
              },
              Object {
                "fontSize": 16.666666666666668,
              },
              Object {
                "color": "#ffffff",
                "fontFamily": "fira-sans",
              },
            ]
          }
        >
          Secs
        </Text>
      </View>
    </View>
  </View>
</View>
`;
